the-state-machine/python/the_state_machine/the_state_machine_stack.py

```diff
@@ -17,17 +17,19 @@ def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
 
         # The first thing we need to do is see if they are asking for pineapple on a pizza
         pineapple_check_lambda = _lambda.Function(self, "pineappleCheckLambdaHandler",
-                                                  runtime=_lambda.Runtime.NODEJS_12_X,
-                                                  handler="orderPizza.handler",
-                                                  code=_lambda.Code.from_asset("lambda_fns"),
+                                                  runtime=_lambda.Runtime.NODEJS_12_X, # execution environment
+                                                  handler="orderPizza.handler", # file is "orderPizza", function is "handler"
+                                                  code=_lambda.Code.from_asset("lambda_fns"), # code loaded from the "lambda_fns" directory
                                                   )
 
         # Step functions are built up of steps, we need to define our first step
-        order_pizza = step_fn.Task(self, 'Order Pizza Job',
-                                   task=step_fn_tasks.InvokeFunction(pineapple_check_lambda),
-                                   input_path='$.flavour',
-                                   result_path='$.pineappleAnalysis'
-                                   )
+        # Refactored code because sfn.Task is Depecrated right now. Using StepFunctionsTaks.LambdaInvoke instead
+        # https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions.Task.html
+        order_pizza = step_fn_tasks.LambdaInvoke(self, 'Order Pizza Job',
```

Comment:
This piece of code was obsolete (https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions.Task.html). I changed it to the new implementation using service as invocation (https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions-tasks.LambdaInvoke.html)